### ISSUE TITLE
Support 0d arrays in cuda

### ISSUE THREAD

**Issue by eric-wieser** [numba/numba#5358]
## Feature request
I'm curious what the intent behind these lines is:

https://github.com/numba/numba/blob/0aaf62d830e7a9d1dc5ffbf710a7cd38caa52d1a/numba/cuda/cudadrv/devicearray.py#L681-L686

Converting 0d arrays to 1d arrays often makes it harder to reason about the behavior of array code.

Following on from https://github.com/numba/numba/pull/5343#discussion_r387971072, where I spotted this accidentally - supposedly it is tested for in ` TestCudaNDArray.test_device_array_interface`.

**Comment by sklam** [comment]
IIRC, the reason is related to 0D array has no `ary.ctypes`.

**Comment by eric-wieser** [comment]
Not entirely sure what you mean by that:
```
In [10]: np.array(1).ctypes
Out[10]: <numpy.core._internal._ctypes at 0x1bd01a8d520>
```

**Comment by eric-wieser** [comment]
Fixed in #5359
